### PR TITLE
Refactor the filter in the frontend to prevent repeated BE calls.

### PR DESCRIPTION
**Pull request by WryanCartie**
Refactored the FE in such a way that the filter happen only in the FE by placing the data in localStorage and filtering it using js functions. Removed all FE and middleware calls to the BE.

### CODE REVIEW

File: src/renderers/PartialViewRenderers/region/componentsInfo.js [hunk 1]
```diff
@@ -1,3 +1,13 @@
+//START CONST LIST
```
Comment (Phobez) [inline src/renderers/PartialViewRenderers/region/componentsInfo.js:1]: You don't need this comment

File: src/renderers/PartialViewRenderers/region/componentsInfo.js [hunk 2]
```diff
@@ -1,3 +1,13 @@
...
+
+const USED = 'used'
+const UNUSED = 'unused'
+const ACTIVATED = 'activated'
+const UNACTIVATED = 'unactivated'
+const COMPONENT_LIST = 'componentList'
+
+//END COSNT LIST
```
Comment (Phobez) [inline src/renderers/PartialViewRenderers/region/componentsInfo.js:9]: This either

File: src/renderers/PartialViewRenderers/region/componentsInfo.js [hunk 3]
```diff
@@ -74,6 +84,9 @@ $(function () {
 function getComponentsInfo() {
     ipcRenderer.send('Component:getComponentList', parseInt(getProcessArgObj()));
     ipcRenderer.once('Component:getComponentListOK', (e, res) => {
+        localStorage.setItem(COMPONENT_LIST, JSON.stringify(res));
+
+
```
Comment (Phobez) [inline src/renderers/PartialViewRenderers/region/componentsInfo.js:89]: Extra space?

File: src/renderers/PartialViewRenderers/region/componentsInfo.js [hunk 4]
```diff
@@ -170,65 +183,36 @@ function getFacilitiesList() {
...
-        componentReceiverEvent = 'Component:getUsedComponentListOK';
-    } else if (usedComponent === 'unused') {
-        componentGetterEvent = 'Component:getUnusedComponentList';
-        componentReceiverEvent = 'Component:getUnusedComponentListOK';
+function filterComponents() {
+    let usedComponentInput = $('input[name=usedComponentDisplay]:checked').val();
+    let activatedComponentInput = $('input[name=activatedComponentDisplay]:checked').val();  
+    let componentList = JSON.parse(localStorage.getItem(COMPONENT_LIST))
+
+    if(usedComponentInput === USED){
```
Comment (Phobez) [inline src/renderers/PartialViewRenderers/region/componentsInfo.js:200]: This code is confusing, if the used component input is used?

Comment (Phobez) [reply src/renderers/PartialViewRenderers/region/componentsInfo.js:200]: If I'm not misunderstanding this, this is the filter right? How about calling it that? Similar to the other one

File: src/renderers/PartialViewRenderers/region/componentsInfo.js [hunk 5]
```diff
@@ -170,65 +183,36 @@ function getFacilitiesList() {
...
+        componentList = componentList.filter(component=> component.facilityId !== null)
+    }else
```
Comment (Phobez) [inline src/renderers/PartialViewRenderers/region/componentsInfo.js:202]: Reformat this please

File: src/renderers/PartialViewRenderers/region/componentsInfo.js [hunk 6]
```diff
@@ -170,65 +183,36 @@ function getFacilitiesList() {
...
-                              (!showActivated && componentData.activationTime > 0);
-        let doShow = ($(this).text().toLowerCase().indexOf(searchString) > -1) && doShowActivated;
-        if (doShow) {
-            $(this).show();
-        } else {
-            hiddenRowIds.push(`#${row.id}`);
-            $(this).hide();
-        }
-    });
+    if(activatedComponentInput === ACTIVATED){
```
Comment (Phobez) [inline src/renderers/PartialViewRenderers/region/componentsInfo.js:207]: Same here, a more descriptive name perhaps? Feel free to chat me to brainstorm name ideas :v

File: src/renderers/PartialViewRenderers/region/componentsInfo.js [hunk 7]
```diff
@@ -170,65 +183,36 @@ function getFacilitiesList() {
...
+        componentList = componentList.filter(component => component.activationTime === 0)
+    }else
+    if(activatedComponentInput === UNACTIVATED){
+
```
Comment (Phobez) [inline src/renderers/PartialViewRenderers/region/componentsInfo.js:211]: Unnecessary line

File: src/renderers/PartialViewRenderers/region/componentsInfo.js [hunk 8]
```diff
@@ -170,65 +183,36 @@ function getFacilitiesList() {
...
-        if (doShow) {
-            $(this).show();
-        } else {
-            hiddenRowIds.push(`#${row.id}`);
-            $(this).hide();
-        }
-    });
+    if(activatedComponentInput === ACTIVATED){
+        componentList = componentList.filter(component => component.activationTime === 0)
+    }else
```
Comment (Phobez) [inline src/renderers/PartialViewRenderers/region/componentsInfo.js:209]: Reformat this too

**Review by Phobez — CHANGES_REQUESTED** [review]

File: src/renderers/PartialViewRenderers/region/componentsInfo.js [hunk 9]
```diff
@@ -170,65 +178,36 @@ function getFacilitiesList() {
...
-        if (doShow) {
-            $(this).show();
-        } else {
-            hiddenRowIds.push(`#${row.id}`);
-            $(this).hide();
-        }
-    });
+    if(componentActivationFilterMode === ACTIVATED){
+        componentList = componentList.filter(component => component.activationTime === 0)
+    }else
```
Comment (Phobez) [inline src/renderers/PartialViewRenderers/region/componentsInfo.js:204]: You haven't fixed this formatting?

File: src/renderers/PartialViewRenderers/region/componentsInfo.js [hunk 10]
```diff
@@ -1,361 +1,373 @@
...
+  addUpdateComponent_handler();
+  //handle delete component events
+  deleteComponent_handler();
+  //handle events from main page
+  pageMain_eventHandler();
+  //handle events from child component modal
+  mdlChildComponents_eventHandler();
+  //handle events from btnOpenBulkInsertComponents
+  btnOpenBulkInsertComponents_eventHandler();
+  //handle events from bulk insert page
```
Comment (Phobez) [inline src/renderers/PartialViewRenderers/region/componentsInfo.js:84]: Could you remove all the comments

File: src/renderers/PartialViewRenderers/region/componentsInfo.js [hunk 11]
```diff
@@ -1,361 +1,373 @@
...
+  //     if (!isNaN(activationTime)) {
+  //         if (activationTime > 0) {
+  //             $('#selFacility').prop('disabled', 'disabled');
+  //             $('#selFacility').val('');
+  //         }
+  //         else {
+  //             $('#selFacility').prop('disabled', false);
+  //         }
+  //     }
+  // })
```
Comment (Phobez) [inline src/renderers/PartialViewRenderers/region/componentsInfo.js:303]: Could you remove this comment

**Review by Phobez — COMMENTED** [review]
Did you lint it?

File: src/renderers/PartialViewRenderers/region/componentsInfo.js (hunk 10 above)
Comment (WryanCartie) [reply src/renderers/PartialViewRenderers/region/componentsInfo.js:84]: Sure aaaa

Comment (Phobez) [reply src/renderers/PartialViewRenderers/region/componentsInfo.js:84]: You haven't removed these comments

Comment (WryanCartie) [reply src/renderers/PartialViewRenderers/region/componentsInfo.js:84]: Oh I didn't realize that was comments outside of my code scope, do you want me to just remove all comments in general?

Comment (Phobez) [reply src/renderers/PartialViewRenderers/region/componentsInfo.js:84]: Nah, let's keep it local just here first

**Review by Phobez — APPROVED** [review]

**Pull request merged**